tools/code-owners-parser/CodeOwnersParser/TeamUserHolder.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Threading.Tasks;
+
+namespace Azure.Sdk.Tools.CodeOwnersParser
+{
+    public class TeamUserHolder
+    {
+        private string TeamUserStorageURI { get; set; } = DefaultStorageConstants.DefaultStorageURI;
+        private Dictionary<string, List<string>>? _teamUserDict = null;
+
+        public Dictionary<string, List<string>> TeamUserDict
+        {
+            get
+            {
+                if (_teamUserDict == null)
+                {
+                    _teamUserDict = GetTeamUserData();
+                }
+                return _teamUserDict;
+            }
+            set
+            {
+                _teamUserDict = value;
+            }
+        }
+
+        public TeamUserHolder(string? teamUserStorageURI)
+        {
+            if (!string.IsNullOrWhiteSpace(teamUserStorageURI))
+            {
+                TeamUserStorageURI = teamUserStorageURI;
+            }
+        }
+
+        private Dictionary<string, List<string>> GetTeamUserData()
+        {
+            if (null == _teamUserDict)
+            {
+                string rawJson = FileHelpers.GetFileOrUrlContents(TeamUserStorageURI);
+                var list = JsonSerializer.Deserialize<List<KeyValuePair<string, List<string>>>>(rawJson);
+                if (null != list)
+                {
+                    return list.ToDictionary((keyItem) => keyItem.Key, (valueItem) => valueItem.Value);
+                }
+                Console.WriteLine($"Error! Unable to deserialize json team/user data. rawJson={rawJson}");
```

Comment:
Probably worth including the TeamUserStorageURI as part of the error message as well.

Reply:
I'll add this in another commit before I merge.